rivertest/worker_test.go

```diff
@@ -326,6 +327,88 @@ func TestWorker_Work(t *testing.T) {
 		require.ErrorContains(t, err, "failed to insert job: tx is closed")
 		require.Nil(t, res)
 	})
+
+	t.Run("WorkUsesHooks", func(t *testing.T) {
```

Comment:
These test cases verify that (1) hooks/middlewares are used in the test worker, and (2) that a hook/middleware with base service is initialized. We weren't previously testing for (1), so I wrapped them up together.